### PR TITLE
Feature : Rename project

### PR DESCRIPTION
**Pull request by justinehell**
- Rename a project
- Create a blank project with custom name
- Create project from template with custom name

### CODE REVIEW

**Review by k-y0u — DISMISSED** [review]

File: src/components/dialog/NewProjectDialog.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const router = useRouter();
+
+/**
+ * Create project on fs, init project on git,
+ * close NewProject and redirect to project page.
+ */
+async function createProject(projectId) {
+  const project = { id: projectId };
+
+  await initProject(project);
```
Comment (k-y0u) [inline src/components/dialog/NewProjectDialog.vue:37]: no error management

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: tests/unit/composables/QuasarFieldRule.spec.js [hunk 2]
```diff
@@ -174,4 +175,15 @@ describe('Test composable: InputRule', () => {
       expect(isNumberTooBig(t, 4, 3)).toEqual(key);
     });
   });
+
+  describe('Test function: isUniqueProjectName', () => {
+    it('should return true without duplicated project', () => {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply tests/unit/composables/QuasarFieldRule.spec.js:194]: done

File: tests/unit/composables/QuasarFieldRule.spec.js [hunk 3]
```diff
@@ -174,4 +175,15 @@ describe('Test composable: InputRule', () => {
...
+      expect(isUniqueProjectName(t, ['projectId'], 'test')).toBe(true);
+    });
+
+    it('should return string error message with duplicated project', () => {
```
[the remark this answers is not in the document]
Comment (justinehell) [reply tests/unit/composables/QuasarFieldRule.spec.js:198]: done

File: src/components/dialog/NewProjectDialog.vue (hunk 1 above)
Comment (justinehell) [reply src/components/dialog/NewProjectDialog.vue:37]: done

File: src/components/form/NewProjectForm.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+const { t } = useI18n();
+const projectName = ref('');
+
+/**
+ * Emit new project name and send positive notification.
+ */
+function onSubmit() {
+  emit('project:create', projectName.value);
+
+  Notify.create({
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/form/NewProjectForm.vue:53]: moved inside NewProjectDialog

**Review by k-y0u — DISMISSED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**